src/se/vidstige/jadb/JadbConnection.java

```diff
@@ -25,6 +25,20 @@ public Transport createTransport() throws IOException {
         return new Transport(new Socket(host, port));
     }
 
+    public void connect(String ip, int port) throws IOException, JadbException {
+        Transport main = this.createTransport();
+        main.send("host:connect:" + ip + ":" + port);
+        main.verifyResponse();
+        main.close();
+    }
```

Comment:
Let's improve the name of this method. Perhaps `restartDeamon` to match the adb command line utility?

Also is the first parameter really ip? It would seem it's the serial number of the device?